agent/cache/cache.go

```diff
@@ -554,9 +554,14 @@ func (c *Cache) fetch(key string, r getOptions, allowNew bool, attempt uint, ign
 		return ch
 	}
 
-	// If we already have an entry and it is actively fetching, then return
-	// the currently active waiter.
+	// If we already have an entry, it is actively fetching, and we don't need
+	// to revalidate the cache, then return the currently active waiter.
 	if ok && entry.Fetching {
+		if r.Info.MustRevalidate {
+			ch := make(chan struct{})
+			close(ch)
+			return ch
+		}
```

Comment:
Not too familiar with this part of the code but this seems to solve the problems and not fail any tests.

Basically, scenario is:
```
1. GET http://localhost:8500/v1/agent/connect/ca/leaf/something?index=35
2. Cancel blocking query
```
I believe this would leave entry in a `Fetching` state until CA was rotated.

Any non-blocking queries would fetch this entry then be forced to block anyways since the `entry.Waiter` would not send until the CA rotation.

This patch will force any non-BQ requests with `MustRevalidate` where the preceding request left `entry` in a `Fetching` state to immediately `goto RETRY_GET` and return the cached entry instead.

I'd appreciate some 👀  to check if this is a sane solution